pyrules/conditions.py

```diff
@@ -0,0 +1,238 @@
+        else:
+            self._expr = ()
+        for k, v in kwargs.iteritems():
+            self._expr = self.combine(C(k, v))._expr
+
+    def clone(self):
+        """
+        Create a new C-expression that is a copy of this one.
+        """
+        new = self.__class__()
+        new._expr = self._expr
+        return new
+            
+    def combine(self, other, op=default):
+        """
+        Combine another C-expression with this using a logical operator.
+        """
+        assert(isinstance(other, C))
+        self._expr = (
+            op, self._expr, other._expr, False) if self._expr else other._expr
+        return self
+
+    def apply(self, context):
+        """
+        Apply evaluation context to this C-expression. Context should be a dict
+        or dict-like object.
+        """
+        return self._apply_expr(context, self._expr)
+
+    def _apply_expr(self, context, expr):
+        if isinstance(expr[1], tuple):
+            return self.OPS_MAP[expr[0]](
+                self._apply_expr(context, expr[1]),
+                self._apply_expr(context, expr[2]))
+        else:
+            # expr[2] holds negation flag, that's why we xor with it.
+            return ExpressionHandler().evaluate(
+                expr[0], context, expr[1]) ^ expr[2]
+
+    def __repr__(self):
+        return '<C: {}>'.format(self._to_str(self._expr)).encode('utf-8')
+
+    def _to_str(self, val):
+        if not val:
+            return u'N/A'
+        elif isinstance(val[1], tuple):
+            return u'{}({} {} {})'.format(
+                'NOT' if val[-1] else '', self._to_str(val[1]), val[0],
+                self._to_str(val[2]))
+        else:
+            return u'{} {}= {}'.format(
+                val[0], '!' if val[-1] else '=', val[1])            
+
+    def __and__(self, other):
+        return self.clone().combine(other, self.AND)
+
+    def __or__(self, other):
+        return self.clone().combine(other, self.OR)
+
+    def __xor__(self, other):
+        return self.clone().combine(other, self.XOR)
+
+    def __invert__(self):
+        new = self.clone()
+        new._expr = tuple(self._expr[:-1]) + (not self._expr[-1],)
+        return new
+
+    def __eq__(self, other):
+        if not issubclass(other.__class__, C):
+            return False
+        return self._expr == other._expr
+
+
+class ExpressionError(Exception):
+    pass
+
+        
+class ExpressionHandler(object):
+    handlers = [
+        'gt', 'lt', 'gte', 'lte', 'eq', 'neq', 'contains', 'icontains', 'bool']
+
+    def evaluate(self, expression, context, arg):
+        bits = expression.split('__')
+        last = len(bits) - 2
+        bit = bits[0]
+        try:
+            cur = context[bit]
+        except KeyError:
+            raise ExpressionError(
+                'Can\'t resolve "{}" bit in expression: "{}"'.format(
+                    bit, expression))
+        for i, bit in enumerate(bits[1:]):
```

Comment:
This is for supporting nested objects or dictionaries. There's no limit to number of bits, not sure why you're confused that there can be only one level. There's an example in tests that contains some nested lookups.

In fact, I think we could replace "__" with "." and then it would work pretty much like django's template variable resolution (maybe it's even better if we do it - for text rules).